go-controller/hybrid-overlay/pkg/controller/overlay_windows.go

```diff
@@ -256,6 +258,11 @@ func GetExistingNetwork(networkName string, expectedAddressPrefix string, expect
 		}
 	}
 
+	if existingNetwork != nil {
+		// the named network already exists but the nodes sunbet or GW has changed
```

Comment:
Shouldn't you actually check the existing network with the node metadata to confirm this? This way we wouldn't always delete the existing network if the node object matches the existing network configuration. And it would automatically pick up the case we are trying to fix i.e. the node object does not exist but the network does. WDYT?

Reply:
isn't that what we are doing? We are grabbing the existing network by name and checking the subnet and gateway address?

Reply:
Ah yes, you are right. Missed the check above. Sorry for the noise.